### PR TITLE
Add SetTriggering functionality and example

### PR DESCRIPTION
**Pull request by marcus-orchard**
Basic changes to allow monitored item triggering for a given subscription with an example.

### CODE REVIEW

File: examples/trigger/trigger.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+	"github.com/gopcua/opcua"
+	"github.com/gopcua/opcua/debug"
+	"github.com/gopcua/opcua/id"
+	"github.com/gopcua/opcua/ua"
+)
+
+func main() {
+	var (
+		endpoint      = flag.String("endpoint", "opc.tcp://morchard-mbp1.local:53530/OPCUA/SimulationServer", "OPC UA Endpoint URL")
```
Comment (kung-foo) [inline examples/trigger/trigger.go]: odds that my laptop is called `morchard-mbp1` is quite low :)

**Review by kung-foo — CHANGES_REQUESTED** [review]

File: subscription.go [hunk 2]
```diff
@@ -112,6 +112,22 @@ func (s *Subscription) Unmonitor(monitoredItemIDs ...uint32) (*ua.DeleteMonitore
 	return res, err
 }
 
+func (s *Subscription) SetTriggering(triggeringItemId uint32, add, remove []uint32) (*ua.SetTriggeringResponse, error) {
```
Comment (kung-foo) [inline subscription.go]: should be `triggeringItemID`

File: examples/trigger/trigger.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+	go sub.Run(ctx) // start Publish loop
+
+	triggeringServerID, triggeredServerID := subRes.Results[0].MonitoredItemID, subRes.Results[1].MonitoredItemID
+	tRes, err := sub.SetTriggering(triggeringServerID, []uint32{triggeredServerID}, []uint32{})
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Print(tRes)
```
Comment (kung-foo) [inline examples/trigger/trigger.go]: This just dumps `&{0xc00020f6d0 [The monitoring item id does not refer to a valid monitored item. StatusBadMonitoredItemIDInvalid (0x80420000)] [] [] []}`. Is that useful?

File: examples/trigger/trigger.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+	subRes, err := sub.Monitor(ua.TimestampsToReturnBoth, miCreateRequests...)
+	if err != nil || subRes.Results[0].StatusCode != ua.StatusOK {
+		log.Fatal(err)
+	}
+
+	go sub.Run(ctx) // start Publish loop
+
+	triggeringServerID, triggeredServerID := subRes.Results[0].MonitoredItemID, subRes.Results[1].MonitoredItemID
+	tRes, err := sub.SetTriggering(triggeringServerID, []uint32{triggeredServerID}, []uint32{})
```
Comment (kung-foo) [inline examples/trigger/trigger.go]: last param (`remove`) can be `nil`.

File: examples/trigger/trigger.go (hunk 1 above)
Comment (marcus-orchard) [reply examples/trigger/trigger.go]: lol, dang. ok, most of these look like they suffer from classic late Friday checkin syndrome. my apologies, I updated this to match the default in the other examples.

File: examples/trigger/trigger.go (hunk 3 above)
Comment (marcus-orchard) [reply examples/trigger/trigger.go]: i changed it to check the status and throw an error if not ok instead.

File: subscription.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcus-orchard) [reply subscription.go]: that makes me more comfortable anyways, was trying to live by the "when in rome..." adage when I skipped the commenting originally.

**Pull request merged**